source/features/default-branch-button.tsx

```diff
@@ -24,7 +24,11 @@ async function init(): Promise<false | void> {
 		url = replaceBranch(currentBranch, defaultBranch);
 	}
 
-	const branchSelector = (await elementReady('#branch-select-menu'))!;
+	const branchSelector = (await elementReady('#branch-select-menu'));
+	if (!branchSelector) { // File history page
+		return;
+	}
+
```

Comment:
If it doesn't work there, `exclude` it

Folders don't work either: https://github.com/sindresorhus/refined-github/commits/master/source

I'm not sure why `pageDetect.isRepoCommitList` is listed here. Where is this supposed to work? Can you double check?

Reply:
#2946

Reply:
The reason I did it this way was if I do it in exclude it may not catch it since the waitForDomReady is set to false

Reply:
I see, this only works on `isRepoCommitList` **unless** you're in a folder/file

https://github.com/sindresorhus/refined-github/commits/6c46766237ccbf8a8c23ba07aca1e48a11a71b4b

So instead of stopping it here, stop it at the very beginning of `init`:

```js
if parseRoute() filepath == undefined
  return false
```

Reply:
Awesome idea!